src/lua/LuaPiGui.cpp

```diff
@@ -493,6 +493,55 @@ static int l_pigui_get_scroll_y(lua_State *l)
 	return 1;
 }
 
+/*
+ * Function: plotHistogram
+ *
+ * Make a histogram
+ *
+ * > ui.plotHistogram(label, data, display_count, overlay, y_min, y_max, graph_size)
+ *
+ * Example:
+ *
+ * Plots 5 bars, with height 17,3,15,5, and 8, respectively
+ *
+ * > ui.plotHistogram("Test", {17,3,15,5,8}, 0)
+ *
+ * Parameters:
+ *
+ *   label - string, text on button
+ *   data - table of values
+ *   display_count - number of pionts to plot, 0 for all
+ *   overlay_text - optional title string, to put on histogram
+ *   y_min - optional float, setting min y-value displayed
+ *   y_max - optional float, setting max y-value displayed
+ *   graph_size - optional Vector2, defining size
+ *
+ */
+static int l_pigui_plot_histogram(lua_State *l)
+{
+	PROFILE_SCOPED()
+	std::string label = LuaPull<std::string>(l, 1);
+	LuaTable vals = LuaTable(l, 2);
+	float values[vals.Size()];
+	float max = FLT_MIN;
+	float min = FLT_MAX;
+	for (int i = 1; i <= vals.Size(); i++) {
+		values[i - 1] = vals.Get<int>(i);
+		if (values[i - 1] > max)
+			max = values[i - 1];
+		if (values[i - 1] < min)
+			min = values[i - 1];
+	}
+	int display_count = LuaPull<int>(l, 3);
+	const char *overlay_text = LuaPull<const char *>(l, 4, NULL);
+	float y_min = LuaPull<float>(l, 5, min);
+	float y_max = LuaPull<float>(l, 6, max);
+	ImVec2 graph_size = LuaPull<ImVec2>(l, 7, ImVec2(80, 80));
```

Comment:
If the calling code isn't passing a value for the histogram, let ImGui size it appropriately instead of constraining it to a 80px by 80px square.

```suggestion
	ImVec2 graph_size = LuaPull<ImVec2>(l, 7, ImVec2(0, 0));
```

Reply:
I did try the 0,0 option, but thought the window became strangely elongated, see screenshot in previous comment